File: notes/1248-Fall2024/155E/code/cheer.c
/**
 * TODO: author header: name, date, purpose
 *
 */
#include <stdlib.h>
#include <stdio.h>

int main(int argc, char **argv) {

    if(argc != 4) {
        printf("ERROR: provide a b n\n");
        exit(1);
    }

    int a = atoi(argv[1]);
    int b = atoi(argv[2]);
    int n = atoi(argv[3]);

    if(a < 2 || b < 2 || n < 1) {
        printf("ERROR; invalid input\n");
        exit(2);
    }

    for(int i=1; i<=n; i++) {
        if(i % a == 0 && i % b == 0) {
            printf("Go Nebraska!\n");
        } else if(i % a == 0) {
            printf("Go Huskers\n");
        } else if(i % b == 0) {
            printf("Go Mavs\n");
        } else {
            printf("%d\n", i);
        }
    }

    return 0;
}

File: notes/1248-Fall2024/155H/code/Cheer.java
package unl.soc;

public class Cheer {

	public static void main(String args[]) {

		if (args.length != 3) {
			System.err.println("ERROR: provide a b n");
			System.exit(1);
		}

		int a = Integer.parseInt(args[0]);
		int b = Integer.parseInt(args[1]);
		int n = Integer.parseInt(args[2]);

		if (a < 2 || b < 2 || n <= 0) {
			System.err.println("ERROR: provide valid inputs");
			System.exit(2);
		}

		for (int i = 1; i <= n; i++) {
			if (i % (a * b) == 0) {
				System.out.println("Go Nebraska!");
			} else if (i % a == 0) {
				System.out.println("Go Huskers!");
			} else if (i % b == 0) {
				System.out.println("Go Mavs!");
			} else {
				System.out.println(i);

			}
		}

	}

}
